azure-mgmt-resources/src/test/java/com/azure/management/resources/core/TestBase.java

```diff
@@ -224,6 +234,44 @@ public void afterTest() throws IOException {
         interceptorManager.finalizeInterceptor();
     }
 
+    protected HttpClient generateHttpClientWithProxy(ProxyOptions proxyOptions) {
+        NettyAsyncHttpClientBuilder clientBuilder = new NettyAsyncHttpClientBuilder();
+        if (proxyOptions != null) {
+            clientBuilder.proxy(proxyOptions);
+        } else {
+            try {
+                System.setProperty("java.net.useSystemProxies", "true");
+                List<Proxy> proxies = ProxySelector.getDefault().select(new URI(AzureEnvironment.AZURE.getResourceManagerEndpoint()));
+                if (!proxies.isEmpty()) {
+                    for (Proxy proxy : proxies) {
+                        if (proxy.address() instanceof InetSocketAddress) {
+                            switch (proxy.type()) {
+                                case HTTP:
+                                    return clientBuilder.proxy(new ProxyOptions(ProxyOptions.Type.HTTP, (InetSocketAddress) proxy.address())).build();
+                                case SOCKS:
+                                    return clientBuilder.proxy(new ProxyOptions(ProxyOptions.Type.SOCKS5, (InetSocketAddress)proxy.address())).build();
+                                default:
+                            }
+                        }
+                    }
+                }
+                String host = null;
+                int port = 0;
+                if (System.getProperty("https.proxyHost") != null && System.getProperty("https.proxyPort") != null) {
+                    host = System.getProperty("https.proxyHost");
+                    port = Integer.parseInt(System.getProperty("https.proxyPort"));
+                } else if (System.getProperty("http.proxyHost") != null && System.getProperty("http.proxyPort") != null) {
+                    host = System.getProperty("https.proxyHost");
+                    port = Integer.parseInt(System.getProperty("http.proxyPort"));
+                }
+                if (host != null) {
+                    clientBuilder.proxy(new ProxyOptions(ProxyOptions.Type.HTTP, new InetSocketAddress(host, port)));
```

Comment:
Why http/https both map to HTTP, not HTTPS?

Reply:
The proxy type has http only.